src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -158,6 +168,35 @@ public void initialize() {
         }
     }
 
+    /**
+     * Helper function to block "drop all on start" from happening on undesired databases
+     */
+    public boolean allowDropAllOnStart() {
+        boolean allowedProfile = Arrays.stream(environment.getActiveProfiles()).anyMatch(e -> e.contains("dev")
+            || e.contains("test") || e.contains("int"));
+
+        List<String> dataProjectNoDrop = migrateConfiguration.getDataProjectNoDropAll();
+        boolean noDeleteDataProject = dataProjectNoDrop != null &&
+            dataProjectNoDrop.contains(resourceConfiguration.getSingleDataProjectId());
+        return allowedProfile && !noDeleteDataProject;
+    }
+
+    /**
+     * Helper function to set "drop all on start" only when two conditions are met:
+     * 1. We allow drop all on start according to allowDropAllOnStart()
+     * 2. "drop all on start" configuration set to true
+     */
+    public boolean getDropAllOnStart() {
+        boolean allowDropAllOnStart = allowDropAllOnStart();
+        boolean dropAllOnStartConfig = migrateConfiguration.getDropAllOnStart();
+        boolean dropAllOnStart = allowDropAllOnStart && dropAllOnStartConfig;
+        logger.info("DropAllOnStart? {} => allowDropAllOnStart = {} && dropAllOnStartConfig = {}",
```

Comment:
This is probably DEBUG not INFO. Once the code is working correctly, will we need to know what these values are?

Reply:
I'd like to keep this at INFO for now. It only happens once per startup and I think it's helpful to be able to easily see which setting is allowing/preventing the drop all.